src/lib/three-d-secure.js

```diff
@@ -1,152 +1,47 @@
 'use strict';
 
-var fs = require('fs');
-var classList = require('@braintree/class-list');
+var assign = require('./assign').assign;
 var threeDSecure = require('braintree-web/three-d-secure');
-var Promise = require('./promise');
 
-function ThreeDSecure(client, merchantConfiguration, cardVerificationString) {
+function ThreeDSecure(client, merchantConfiguration) {
   this._client = client;
   this._config = merchantConfiguration;
-  this._modal = this._setupModal(cardVerificationString);
 }
 
 ThreeDSecure.prototype.initialize = function () {
   var self = this;
 
   return threeDSecure.create({
-    client: this._client
+    client: this._client,
+    version: 2
   }).then(function (instance) {
     self._instance = instance;
   });
 };
 
-ThreeDSecure.prototype.verify = function (nonce) {
-  var self = this;
-
-  this._revealModal();
-
-  return Promise.all([
-    this._waitForThreeDSecure(),
-    this._instance.verifyCard({
-      nonce: nonce,
-      amount: this._config.amount,
-      showLoader: false,
-      addFrame: function (err, iframe) {
-        var modalBody = self._modal.querySelector('.braintree-three-d-secure__modal-body');
-
-        iframe.onload = function () {
-          classList.add(modalBody, 'braintree-three-d-secure__frame-active');
-        };
-
-        modalBody.appendChild(iframe);
-      },
-      removeFrame: function () {
-        self._cleanupModal();
-      }
-    }).then(function (payload) {
-      self._resolveThreeDSecure();
-
-      return payload;
-    })
-  ]).then(function (result) {
-    self._cleanupModal();
-
-    return result[1];
-  }).catch(function (err) {
-    self._cleanupModal();
-
-    if (err.type === 'THREE_D_SECURE_CANCELLED') {
-      return Promise.resolve(err.payload);
+ThreeDSecure.prototype.verify = function (payload, merchantProvidedData) {
+  var verifyOptions = assign({
+    amount: this._config.amount
+  }, merchantProvidedData, {
+    nonce: payload.nonce,
+    bin: payload.details.bin,
+    // TODO in the future, we will allow
```

Comment:
do we want to keep this TODO note in?

Reply:
I think so, since it's something we want to add imminently, I'd like it to keep showing up as a warning in eslint.